keras_preprocessing/image.py

```diff
@@ -1554,6 +1554,15 @@ def __init__(self, x, y, image_data_generator,
                 raise ValueError('Invalid subset name:', subset,
                                  '; expected "training" or "validation".')
             split_idx = int(len(x) * image_data_generator._validation_split)
+
+            if len(np.unique(y)) != len(np.unique(y[:split_idx])) or \
```

Comment:
By construction, if np.unique(y) == np.unique(y_val) and np.unique(y) == np.unique(y_train) then np.unique(y_val) == np.unique(y_train). So we only need to check that np.unique(y_train) == np.unique(y_val).

Also to check for equality you can use `np.array_equal`.
`np.array_equal(np.unique(x),np.unique(y))`

In any case, we try to avoid \.